gems/sorbet-runtime/lib/types/props/custom_type.rb

```diff
@@ -24,30 +39,30 @@ def valid?(value)
     #
     # @param [Object] _instance
     # @return An instance of one of T::Configuration.scalar_types
-    def serialize(_instance)
-      raise NotImplementedError.new('Must override in included class')
-    end
+    sig {abstract.params(instance: T.untyped).returns(T.untyped).checked(:never)}
+    def serialize(instance); end
 
     # Given the serialized form of your type, this returns an instance
     # of that custom type representing that value.
     #
-    # @param _mongo_scalar One of T::Configuration.scalar_types
+    # @param scalar One of T::Configuration.scalar_types
     # @return Object
-    def deserialize(_mongo_scalar)
-      raise NotImplementedError.new('Must override in included class')
-    end
+    sig {abstract.params(scalar: T.untyped).returns(T.untyped).checked(:never)}
+    def deserialize(scalar); end
 
+    sig {override.params(_base: Module).void}
     def self.included(_base)
       super
 
       raise 'Please use "extend", not "include" to attach this module'
     end
 
+    sig(:final) {params(val: Object).returns(T::Boolean).checked(:never)}
     def self.scalar_type?(val)
       # We don't need to check for val's included modules in
       # T::Configuration.scalar_types, because T::Configuration.scalar_types
       # are all classes.
-      klass = val.class
+      klass = T.let(val.class, T.nilable(Class))
```

Comment:
`class Class < Module` in Ruby. Should this be `T.nilable(Module)`? Would it matter?

Reply:
I think it can't be a Module that's not a class, because it comes from `.class` and then from `.superclass`.